include/eve/detail/function/simd/common/friends.hpp

```diff
@@ -135,7 +142,24 @@ namespace eve::detail
   }
 
   //================================================================================================
-  template<real_simd_value Wide>
+  template<simd_value Wide>
+  EVE_FORCEINLINE auto self_eq(Wide const& v,Wide const& w) noexcept
+  requires( kumi::product_type<element_type_t<Wide>> )
+  {
+    if constexpr( detail::tag_dispatchable<tag::is_equal_,Wide,Wide> )
+    {
+      return tagged_dispatch(tag::is_equal_{}, v, w);
+    }
+    else
+    {
+      auto zipped = kumi::zip(v.storage(),w.storage());
+      return kumi::fold_right ( [](auto acc , auto e) { return acc && ( get<0>(e) == get<1>(e) ); }
+                              , zipped, true
+                              );
+    }
```

Comment:
Does this use wide equality?

Reply:
each member is compared using the == of the wide beneath yes